### PR TITLE
Port over missing debug info driver functionality from the C++ implem…

### PR DESCRIPTION
**Pull request by adrian-prantl**
…entation

### CODE REVIEW

**Comment by adrian-prantl** [comment]
@swift-ci test

**Comment by adrian-prantl** [comment]
This implements the same tests as https://github.com/apple/swift/pull/70315

**Comment by adrian-prantl** [comment]
@swift-ci test

**Comment by adrian-prantl** [comment]
@swift-ci test windows

File: Sources/SwiftDriver/Toolchains/DarwinToolchain.swift [hunk 1]
```diff
@@ -219,6 +219,15 @@ public final class DarwinToolchain: Toolchain {
     }
   }
 
+  public func getDefaultDwarfVersion(targetTriple: Triple) -> UInt8 {
+      if (targetTriple.isMacOSX && targetTriple.version(for: .macOS) < Triple.Version(10, 11, 0)) ||
+        (targetTriple.isiOS && targetTriple.version(
+            for: .iOS(targetTriple._isSimulatorEnvironment ? .simulator : .device)) < Triple.Version(9, 0, 0)) {
+      return 2;
```
Comment (DougGregor) [inline Sources/SwiftDriver/Toolchains/DarwinToolchain.swift:226]: TIL!

**Review by DougGregor — APPROVED** [review]
Looks good, thanks!

**Review by artemcm — APPROVED** [review]

**Pull request merged**